Mobile apps: Clear noise on OAuth2 login page

xavier-lc opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Fixes DOTOBRD-161

## Proposed Changes

* Remove the header and footer links, terms of service text, and the `Back` button on the mobile apps login page.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* As explained on issue, we want to remove potential distractions so the users focus on logging in.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open the live preview
* Go to `/log-in?client_id=2697&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D2697%26redirect_uri%3Dwordpress%253A%252F%252Fwpcom-authorize%26response_type%3Dcode%26scope%3Dglobal%26from-calypso%3D1` (the link you would get if trying to log in on the wordpress mobile app for android)
* Check that the header and footer links, terms of service text, and the `Back` button have been removed.
* Repeat with `/log-in?client_id=11&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D11%26redirect_uri%3Dwordpress%253A%252F%252Fwpcom-authorize%26response_type%3Dcode%26scope%3Dglobal%26from-calypso%3D1` (the link you would get if trying to log in on the wordpress mobile app for iOs)

|Before|After|
|---|---|
|![image](https://github.com/user-attachments/assets/0321cef6-f227-416e-84ec-65ae491b60f2)|![image](https://github.com/user-attachments/assets/dec4e01c-9a62-41ba-a751-62373a9302b5)|

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Remove `Back to ...` button for Android/iOs

Commit:
Remove footer for Android/iOs

Commit:
Remove logged out masterbar for Android/iOs

Commit:
Remove terms for Android/iOs

heavyweight reviewed (commented):
Tested with the link provided and works well 👌
I tried to test with the other client id but I end up on an error page

<img width="1852" alt="Screenshot 2025-06-19 at 12 43 56" src="https://github.com/user-attachments/assets/bc8c1234-7cad-467e-b59f-fcb5b0ef3ec2" />

I could be using a wrong link not sure. Here it is: 
/log-in?client_id=11&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D11%26redirect_uri%3Dwordpress%253A%252F%252Fwpcom-authorize%26response_type%3Dcode%26scope%3Dglobal%26from-calypso%3D1

Commit:
Fix missing `&&`

xavier-lc commented:
> I tried to test with the other client id but I end up on an error page

Thanks for the catch, there was a js error that I've fixed now. I've also added the iOs link on the PR description.

heavyweight reviewed (approved):
LGTM 🤘
